clar2wasm/src/tools.rs

```diff
@@ -476,173 +476,140 @@ fn crosseval(snippet: &str, env: TestEnvironment) -> Result<CrossEvalResult, Kno
-        "value is not the expected {:?}",
-        eval.compiled
-    );
+    Some(eval)
 }
 
-pub fn crosscheck_with_amount(snippet: &str, amount: u128, expected: Result<Option<Value>, Error>) {
-    let eval = match crosseval(
+pub fn crosscheck(snippet: &str, expected: Result<Option<Value>, Error>) {
+    if let Some(eval) = execute_crosscheck(
+        TestEnvironment::new(TestConfig::latest_epoch(), TestConfig::clarity_version()),
         snippet,
+        |_| {},
+    ) {
+        assert_eq!(
+            eval.compiled, expected,
+            "value is not the expected {:?}",
+            eval.compiled
+        );
+    }
+}
+
+pub fn crosscheck_with_amount(snippet: &str, amount: u128, expected: Result<Option<Value>, Error>) {
+    if let Some(eval) = execute_crosscheck(
         TestEnvironment::new_with_amount(
             amount,
             TestConfig::latest_epoch(),
             TestConfig::clarity_version(),
         ),
+        snippet,
+        |_| {},
     ) {
-        Ok(result) => result,
-        Err(_bug) => {
-            return;
-        }
-    };
-
-    eval.compare(snippet);
-
-    assert_eq!(
-        eval.compiled, expected,
-        "value is not the expected {:?}",
-        eval.compiled
-    );
+        assert_eq!(
+            eval.compiled, expected,
+            "value is not the expected {:?}",
+            eval.compiled
+        );
+    }
 }
 
 pub fn crosscheck_compare_only(snippet: &str) {
     // to avoid false positives when both the compiled and interpreted fail,
     // we don't allow failures in these tests
-
-    let eval = match crosseval(
-        snippet,
+    execute_crosscheck(
         TestEnvironment::new(TestConfig::latest_epoch(), TestConfig::clarity_version()),
-    ) {
-        Ok(result) => result,
-        Err(_bug) => {
-            return;
-        }
-    };
-
-    // Note that we interpret first, to catch logical errors early
-    assert!(eval.interpreted.is_ok(), "Interpreted snippet failed");
-    assert!(eval.compiled.is_ok(), "Compiled snippet failed");
-
-    eval.compare(snippet);
+        snippet,
+        |eval| {
+            // Note that we interpret first, to catch logical errors early
+            assert!(eval.interpreted.is_ok(), "Interpreted snippet failed");
+            assert!(eval.compiled.is_ok(), "Compiled snippet failed");
+        },
+    );
 }
 
 pub fn crosscheck_compare_only_with_expected_error<E: Fn(&Error) -> bool>(
     snippet: &str,
     expected: E,
 ) {
-    let eval = match crosseval(
-        snippet,
+    execute_crosscheck(
         TestEnvironment::new(TestConfig::latest_epoch(), TestConfig::clarity_version()),
-    ) {
-        Ok(result) => result,
-        Err(_bug) => {
-            return;
-        }
-    };
-
-    if let Err(e) = &eval.compiled {
-        if !expected(e) {
-            panic!("Compiled snippet failed with unexpected error: {:?}", e);
-        }
-    }
-
-    eval.compare(snippet);
+        snippet,
+        |eval| {
```

Comment:
nitpick: the `eval` here could be called "result" or "evaluated" or something, eval sounds like a verb to me. (also above)

Reply:
Addressed!